qutebrowser/keyinput/eventfilter.py

```diff
@@ -118,8 +120,92 @@ def eventFilter(self, obj: Optional[QObject], event: Optional[QEvent]) -> bool:
+        #      remembering focus for tabs
+        #   anything to unregister? saw some hangs on crash but might be
+        #      because of being a temp basedir
+        # some input examples here https://www.javatpoint.com/html-form-input-types
+        #  <input type="date">: doesn't report as having an input method enabled,
+        #  although the existing heuristics pick it up
+        # if insert_mode_auto_load is false but there is a blinking cursor on
+        # load clicking the scroll bar will enter insert mode
+        assert self._input_method is not None
+
+        new_rect = self._input_method.cursorRectangle()
+        if self._last_seen_rect and self._last_seen_rect.contains(new_rect):
+            print("contains")
+            return
+
+        self._last_seen_rect = new_rect
+
+        # implementation detail: qtwebengine doesn't set anchor for input
+        # fields in a web page, qt widgets do, I haven't found any cases where
+        # it doesn't work yet. Would like to compare with a "get focused thing
+        # and examine" check first and compare across versions.
+        anchor_rect = self._input_method.anchorRectangle()
+        if anchor_rect:
+            print("Not handling because anchor rect is set")
+            return
+
+        focused_window = objreg.last_focused_window()
+        focus_object = QApplication.focusObject()
+        query = None
+
+        if not new_rect and focus_object:
+            # sometimes we get a rectangle changed event and the queried
+            # rectangle is empty but we are still in an editable element. For
+            # instance when pressing enter in a text box on confluence or jira
+            # (including comment on the Qt instance) and tabbing between cells
+            # on https://html-online.com/editor/
+            # Checking ImEnabled helps in these cases.
+            query = QInputMethodQueryEvent(Qt.InputMethodQuery.ImEnabled)
+            QApplication.sendEvent(focus_object, query)
+
+        if new_rect or (query and query.value(Qt.InputMethodQuery.ImEnabled)):
+            log.mouse.debug("Clicked editable element!")
+            if config.val.input.insert_mode.auto_enter:
+                modeman.enter(focused_window.win_id, usertypes.KeyMode.insert,
+                              'click', only_if_normal=True)
+        else:
+            log.mouse.debug("Clicked non-editable element!")
+            if config.val.input.insert_mode.auto_leave:
+                modeman.leave(focused_window.win_id, usertypes.KeyMode.insert,
+                              'click', maybe=True)
+
+
 def init() -> None:
     """Initialize the global EventFilter instance."""
     event_filter = EventFilter(parent=objects.qapp)
     event_filter.install()
     quitter.instance.shutting_down.connect(event_filter.shutdown)
+
+    if config.val.input.better_input_detection:
+        def donothing() -> None:
+            _ = IMEEventHandler(parent=QApplication.instance())
```

Comment:
Why assigning it to a dummy variable? Given it has a `parent=` set, is that needed?